temporal-sdk/src/main/java/io/temporal/worker/WorkerOptions.java

```diff
@@ -519,15 +595,15 @@ public double getMaxWorkerActivitiesPerSecond() {
   }
 
   public int getMaxConcurrentActivityExecutionSize() {
-    return maxConcurrentActivityExecutionSize;
+    return activitySlotSupplier.getMaximumSlots();
   }
 
   public int getMaxConcurrentWorkflowTaskExecutionSize() {
-    return maxConcurrentWorkflowTaskExecutionSize;
+    return workflowSlotSupplier.getMaximumSlots();
   }
 
   public int getMaxConcurrentLocalActivityExecutionSize() {
-    return maxConcurrentLocalActivityExecutionSize;
+    return localActivitySlotSupplier.getMaximumSlots();
```

Comment:
These 3 `getMaxConcurrent*ExecutionSize()` should be zero if the corresponding `maxConcurrent*ExecutionSize` is not set.

Reply:
I think a buidlers setter should only effect one getter

Reply:
Ah, yeah forgot to change those ones.

Reply:
Yeah... I was hoping not to retain the property in the _built_ workflow options since having mutually exclusive stuff represented in the built type is less than good from a type design perspective. It also means the logic for the threadpool size has to get pushed out of the options, but, that's not so bad.